lab-matt/cowsay.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const http = require('http');
+const parseUrl = require('url').parse;
+const parseQuery = require('querystring').parse;
+const cowsay = require('cowsay');
+
+const parseBody = require('./lib/parse-json');
+
+const port = process.argv[2] || 5000;
+
+const server = http.createServer(function(req, res){
+  req.url = parseUrl(req.url);
+  req.url.query = parseQuery(req.url.query);
+
+  if(req.method === 'POST' || 'PUT' || 'DELETE'){
+    parseBody(req).then(function(){
+      if(req.url.pathname === '/'){
+        res.writeHead(200, { 'Content-Type': 'text/plain' });
+        res.write('API Endpoints:\n ' + '/cowsay\n');
+        res.end();
+        return;
+      }
+      if(req.url.pathname === '/cowsay'){
+        res.write(cowsay.say({text: JSON.stringify(req.body.user) || 'Sup Blood'}));
```

Comment:
This does not accept the correct property from JSON the client sends.  It should accept "text".  For example, if the user sends '{"text":"hello"}' the cow should say "hello" (without double quotes).